cuda/test/utils.hpp

```diff
@@ -41,32 +41,22 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/stream.hpp>
 
 
+#include "core/test/gtest/resources.hpp"
 #include "cuda/base/device.hpp"
 
 
 namespace {
 
 
-class CudaEnvironment : public ::testing::Environment {
-public:
-    void TearDown() override { gko::kernels::cuda::reset_device(0); }
-};
-
-testing::Environment* cuda_env =
-    testing::AddGlobalTestEnvironment(new CudaEnvironment);
-
-
 class CudaTestFixture : public ::testing::Test {
 protected:
     CudaTestFixture()
         : ref(gko::ReferenceExecutor::create()),
-#ifdef GKO_TEST_NONDEFAULT_STREAM
-          stream(0),
+          stream(ResourceEnvironment::cuda_device_id),
           exec(gko::CudaExecutor::create(
-              0, ref, std::make_shared<gko::CudaAllocator>(), stream.get()))
-#else
-          exec(gko::CudaExecutor::create(0, ref))
-#endif
+              ResourceEnvironment::cuda_device_id, ref,
+              std::make_shared<gko::CudaAllocator>(), stream.get())),
```

Comment:
it means we always test the nondefault stream, we still need to keep the test for default stream, I think.

Reply:
Agreed, we discussed this already here https://github.com/ginkgo-project/ginkgo/pull/1373#discussion_r1280830747
I need to try and reproduce the issues Marcel had.